src/worker/exporters/node_exporter.py

```diff
@@ -239,24 +298,81 @@ def get_log_level(args):
     elif levelStr == '4' or levelStr == 'DEBUG':
         numeric_log_level = logging.DEBUG
     else:
-        print("Could not understand the specified --log-level '%s'" % (args.loglevel))
+        print(
+            "Could not understand the specified --log-level '%s'" %
+            (args.loglevel))
         args.print_help()
         sys.exit(2)
     return numeric_log_level
 
 
+def init_nvidia_ib_config():
+    global config
+    global GPU_Mapping
+    global IB_Mapping
+    global FIELD_LIST
+    # check if nvidiaVM
+    nvArch = os.path.exists('/dev/nvidiactl')
+    config['command'] = {}
+    if nvArch:
+        config['command']['xid_error'] = "grep 'NVRM: Xid' /var/log/syslog"
+        config['counter']['xid_error'] = {}
+        cmd = 'nvidia-smi -L'
+        result = shell_cmd(cmd, 5)
+        gpuCount = len(result.split('\nGPU'))
+        try:
+            for gpu in range(gpuCount):
+                cmd = 'nvidia-smi -q -g ' + str(gpu) + ' -d ACCOUNTING'
+                result = shell_cmd(cmd, 5)
+                pci = re.search(r"\w+:\w\w:\w\w\.", result).group().lower()
+                pci = pci.replace('.', '')[-10:]
+                GPU_Mapping[pci] = str(gpu)  # pci mapping
+                config['counter']['xid_error'][pci] = {}
+            FIELD_LIST.append('xid_error')
+        except Exception as e:
+            print(e)
+            pass
+
+    # IB mapping
```

Comment:
Would it make sense to move IB mapping into its own function, since it's not part of the nvArch if and it will be needed for AMD GPUs as well?

Reply:
addressed